CHANGELOG.md

```diff
@@ -14,6 +14,7 @@ For experimental package changes, see the [experimental CHANGELOG](experimental/
 ### :bug: (Bug Fix)
 
 * fix(sdk-trace-base): ensure attribute value length limit is enforced on span creation [#4417](https://github.com/open-telemetry/opentelemetry-js/pull/4417) @pichlermarc
+* fix(core): drop unnecessary assignment of HOSTNAME [#4421](https://github.com/open-telemetry/opentelemetry-js/pull/4421) @pichlermarc
```

Comment:
This doesn't sound like a bug fix to me, since it simply removes a no-op which has no user-observable effect.

Reply:
Yep, I was not sure what to categorize it at. Thinking about it, `refactor` may be more accurate. I'll change the entry.